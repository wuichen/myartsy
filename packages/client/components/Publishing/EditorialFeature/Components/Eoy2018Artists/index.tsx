import { Box, color, Flex, media, Sans, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCardContainer } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCard"
import {
  ArticleCards,
  ArticlesWrapper,
} from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import {
  FullLabel,
  ImageSetContainer,
  ImageSetPreview,
  ImgContainer,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  LabelWrapper,
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<EditorialFeaturesProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url
    const caption = imageSection && imageSection.images[0].caption

    return (
      <ArtistHeaderSection
        key={i}
        mb={40}
        flexDirection={["column", "column", "row", "row"]}
      >
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg
          src={src && resize(src, { width: 700 })}
          alignItems="flex-end"
        >
          {caption && (
            <Sans size="1" color="white" px={10} py={5}>
              <div dangerouslySetInnerHTML={{ __html: caption }} />
            </Sans>
          )}
        </ArtistHeaderImg>
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    const {
      article: { sections, layout },
      showTooltips,
    } = this.props
    const isChapterStart =
      sections[i - 1] && sections[i - 1].type === "image_collection"

    return (
      <Box maxWidth={["100%", "75%", "75%"]} ml="auto" px={[20, 0]} key={i}>
        <TextSection size="5" pb={[40, 60]} isChapterStart={isChapterStart}>
          <Text
            html={section.body}
            layout={layout}
            showTooltips={showTooltips}
          />
        </TextSection>
      </Box>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper
        key={i}
        mb={60}
        maxWidth={["100%", "100%", "100%", "75%"]}
      >
        <ImageSetPreview section={section}>
          <CaptionWrapper size={["3", "4"]}>
            <ImageSetCaption
              dangerouslySetInnerHTML={{ __html: section.images[0].caption }}
            />
          </CaptionWrapper>
        </ImageSetPreview>
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article, isMobile, isTablet, isTest } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <AricleWrapper>
        <Nav canFix color={color("black100")} backgroundColor="white">
          <NavBorder />
        </Nav>

        <Box px={[10, 10, 55]} maxWidth={1600} mx="auto">
          <Eoy2018ArticleHeader
            images={headerImages}
            isMobile={isMobile}
            isTablet={isTablet}
            isTest={isTest}
          />

          <ArticleContent py={40}>
            <IntroSection
              alignItems="flex-start"
              flexDirection={["column", "row"]}
              pl={[0, 20]}
            >
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </Box>

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} mb={40} mx="auto">
            <RelatedArticleWrapper>
              <ArticleCards relatedArticles={article.relatedArticles} />
            </RelatedArticleWrapper>
          </Box>
        )}
      </AricleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const AricleWrapper = styled.div`
  ${NavContainer} {
    border-bottom: none;
  }
`

const NavBorder = styled.div`
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  position: absolute;
  top: 100%;
  left: 55px;
  right: 55px;
  max-width: calc(1600px - 110px);
  margin: 0 auto;

  ${media.md`
    left: 10px;
    right: 10px;
  `};
`

const ArticleContent = styled(Box)`
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
    line-height: 1.3em;
    margin: 0;
    font-weight: inherit;

    ${media.sm`
      ${unica("s25")};
      line-height: 1.3em;
    `};
  }
`

export const ArtistHeaderSection = styled(Flex)`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left-width: 0;

  ${media.md`
    height: fit-content;
  `};
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  overflow: hidden;
  min-height: min-content;

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    margin: 0;
    font-weight: inherit;

    ${media.md`
      width: 60%;
      float: right;
      height: 100%;
      border-bottom: none;
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};

    ${media.xs`
      ${unica("s45")};
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
      border-left: none;
      width: 100%;
      height: 50%;
      float: none;
    `};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: ${space(2)}px;
    margin: 0;
    font-weight: inherit;

    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }

    ${media.md`
      width: 40%;
      float: left;
      &:last-child {
        border-left: none;
        border-top: ${BORDER_WIDTH}px solid ${color("purple100")};
      }
    `};

    ${media.xs`
      ${unica("s19")};
      width: 50%;
      float: none;
      &:last-child {
        border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
        border-top: none;
      }
    `};
  }

  ${media.md`
    height: 65vh;
    border-right: none;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ArtistHeaderImg = styled(Flex)<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};

  p {
    text-shadow: 0 0 5px black;
    opacity: 0.6;
  }

  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};

  ${media.md`
    min-height: 70vw;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
    ${media.md`
      padding: 0 20px 20px;
    `};
  }
`

const TextSection = styled(Serif)<{ isChapterStart?: boolean }>`
  ${StyledText} {
    blockquote {
      padding: 0;
    }

    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      ${props =>
        props.isChapterStart &&
        `
        &:first-child {
          text-indent: 0;
        }
    `};
    }
  }
`

const CaptionWrapper = styled(Sans)`
  flex: 1;

  ${media.sm`
    min-width: 50%;
  `};
`

const ImageSetCaption = styled.div`
  height: 100%;
  padding: ${space(2)}px;
  background: white;
  color: black;
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${media.sm`
    border-bottom: none;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ImageSetWrapper = styled(Box)`
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;

  ${ImageSetContainer} {
    display: flex;
    flex-direction: row-reverse;

    ${media.sm`
      flex-direction: column-reverse;
    `};
  }

  ${FullLabel} {
    position: relative;
    display: flex;
    flex-direction: column-reverse;
    flex: 1;
    left: auto;
    bottom: auto;
    max-width: inherit;
    border-radius: 0;
    box-shadow: none;

    &:hover {
      background-color: ${color("purple100")};
    }
    ${media.sm`
      flex-direction: row-reverse;

      ${LabelWrapper} {
        max-width: 50%;
      }
    `};
  }

  ${SlideshowTitle} {
    display: none;
  }

  ${SlideshowCta} {
    flex-direction: column;

    > div {
      padding: 0;
    }
  }

  ${ImgContainer} {
    flex: 3;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

    img {
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    ${media.sm`
      border-right: none;
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};
  }
`

const RelatedArticleWrapper = styled.div`
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${ArticlesWrapper} {
    margin-bottom: 0;
    margin-top: 0;
  }

  ${ArticleCardContainer} {
    border: none;
  }
`
