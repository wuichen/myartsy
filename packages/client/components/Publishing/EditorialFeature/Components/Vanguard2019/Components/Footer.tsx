import { Box, Flex, Sans, Separator, Serif } from "@artsy/palette"
import React from "react"

export const VanguardFooter = () => {
  return (
    <Box background="white" pb={80}>
      <Box
        maxWidth={["100vw", 1000]}
        px={["10vw", "10vw", "10vw", "10vw", 4]}
        mx="auto"
      >
        <Separator mb={4} />
        <Flex flexDirection={["column", "row"]}>
          <Box pr={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Corrections
            </Sans>
            <Serif size="3" element="p">
              {corrections}
            </Serif>
          </Box>
          <Box pl={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Clarifications
            </Sans>
            <Serif size="3" element="p">
              {clarifications}
            </Serif>
          </Box>
        </Flex>
        <Separator mb={2} />
        <Serif size="3">
          Back to <a href="/articles">Artsy Editorial</a>
        </Serif>
      </Box>
    </Box>
  )
}

const corrections = `A previous version of this article referred to Aike gallery as
Aike-Dellarco; McArthur Binion’s show at the Mississippi Museum of
Art, originally scheduled for this fall, is not opening until 2020;
and Melike Kara is Kurdish-German not Turkish-German. The text has
been updated to reflect these changes.`

const clarifications = `While only Beyoncé was named as the buyer of a Derek Fordjour work at
Frieze New York, Jay-Z was actually the buyer; in order to fully
recognize the breadth of galleries that represent the featured
artists, additional gallery names have been added to the text for
Beatriz González, Vivian Suter, and Diane Simpson; while the text on
Henry Taylor previously referred to Eva Presenhuber as his dealer, it
has been updated to reflect that Blum & Poe has been primarily
representing the artist for the past decade; and mention of
SculptureCenter, organizers of Tishan Hsu’s 2020 show at the Hammer,
has been added to clarify their involvement with the exhibition.`
