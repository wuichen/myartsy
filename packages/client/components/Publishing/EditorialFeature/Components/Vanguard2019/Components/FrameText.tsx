import { color, Flex, Sans } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { Media } from "Utils/Responsive"

interface VanguardFrameTextProps {
  isSlideOpen?: boolean
}

export const VanguardFrameText: React.SFC<VanguardFrameTextProps> = props => {
  const { isSlideOpen } = props

  return (
    <>
      <Media at="xs">
        <MobileFrame isSlideOpen={isSlideOpen}>
          <MobileFrameText size="8">The Artsy</MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            Vanguard
          </MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            2019
          </MobileFrameText>
        </MobileFrame>
      </Media>

      {/* Desktop */}
      <Media greaterThan="xs">
        <FrameTextLeft
          isSlideOpen={isSlideOpen}
          size={["12", "12", "14", "16"]}
        >
          Vanguard
        </FrameTextLeft>
        <FrameTextRight
          isSlideOpen={isSlideOpen}
          size={["12", "12", "14", "16"]}
        >
          2019
        </FrameTextRight>
      </Media>
    </>
  )
}

const FrameText = styled(Sans)`
  position: fixed;
  top: 50%;
  text-transform: uppercase;
  z-index: 2;
  mix-blend-mode: difference;
  color: ${color("white100")};
  will-change: color;
`

const FrameTextLeft = styled(FrameText)<{ isSlideOpen?: boolean }>`
  left: 0;
  transform: rotate(-90deg);
  transform-origin: 30% 136%;

  &&& {
    z-index: ${p => (p.isSlideOpen ? 0 : 2)};
  }
`

const FrameTextRight = styled(FrameText)<{ isSlideOpen?: boolean }>`
  right: 0;
  transform: rotate(90deg);
  transform-origin: 60% 60%;

  &&& {
    z-index: ${p => (p.isSlideOpen ? 0 : 2)};
  }
`

const MobileFrame = styled(Flex)<{
  isSlideOpen?: boolean
}>`
  position: fixed;
  flex-direction: column;
  top: 65px;
  left: -10px;
  height: calc(100vh - 150px);
  width: 60px;
  justify-content: space-between;
  z-index: ${p => (p.isSlideOpen ? 0 : 2)};
`

const MobileFrameText = styled(Sans)<{
  isUppercase?: boolean
}>`
  transform: rotate(90deg);
  white-space: nowrap;
  padding: 0 10px;

  ${({ isUppercase }) =>
    isUppercase &&
    `
    text-transform: uppercase;
  `}
`
