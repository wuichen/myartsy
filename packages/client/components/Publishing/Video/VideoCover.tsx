import { Box, Flex, Serif } from "@artsy/palette"
import React, { Component } from "react"
import track, { TrackingProp } from "react-tracking"
import styled from "styled-components"

import { IconVideoPlay } from "Components/Publishing/Icon/IconVideoPlay"
import { ArticleData, MediaData } from "Components/Publishing/Typings"
import { VideoInfoBlock } from "./VideoInfoBlock"

interface VideoCoverProps {
  article: ArticleData
  description?: string
  editDescription?: any
  editTitle?: any
  hideCover?: boolean
  media: MediaData
  playVideo?: () => void
  seriesLink?: string
  seriesTitle?: string
  tracking?: TrackingProp
}

@track()
export class VideoCover extends Component<VideoCoverProps> {
  onPlayClick = () => {
    const { tracking, playVideo } = this.props

    tracking.trackEvent({
      action: "Click",
      label: "Play video",
    })
    playVideo()
  }

  render() {
    const {
      article,
      editDescription,
      editTitle,
      hideCover,
      media,
      seriesLink,
      seriesTitle,
    } = this.props

    return (
      <VideoCoverContainer hideCover={hideCover}>
        <VideoCoverAsset src={media.cover_image_url} />
        <VideoCoverOverlay />

        <VideoCoverInfo
          width="100%"
          alignItems="flex-end"
          pb={[40, 60]}
          px={20}
        >
          <Box maxWidth={1200} mx="auto" pb="12px">
            <Flex>
              <Box width="60px" pr={20} onClick={this.onPlayClick}>
                <IconVideoPlay color="white" />
              </Box>
              <Box>
                <VideoInfoBlock
                  media={media}
                  subTitle={
                    seriesTitle || (article.vertical && article.vertical.name)
                  }
                  subTitleLink={seriesLink}
                  title={article.title}
                  editTitle={editTitle}
                />
              </Box>
            </Flex>

            <Box maxWidth={["100%", "60%"]}>
              <Serif size={["4", "5", "5", "5"]} pt={30}>
                {editDescription || article.description}
              </Serif>
            </Box>
          </Box>
        </VideoCoverInfo>
      </VideoCoverContainer>
    )
  }
}

export const VideoCoverAsset = styled.div<{ src?: string }>`
  background: url(${props => props.src || ""}) no-repeat center center;
  background-size: cover;
  background-color: black;
  position: absolute;
  width: 100%;
  height: 100%;
`

const VideoCoverOverlay = styled.div`
  background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.6));
  position: absolute;
  width: 100%;
  height: 100%;
`

const VideoCoverInfo = styled(Flex)`
  box-sizing: border-box;
  z-index: 1;
  position: absolute;
  width: 100%;
  height: 100%;

  ${IconVideoPlay} {
    height: 60px;
    width: 44px;
    cursor: pointer;
  }
`

export const VideoCoverContainer = styled.div<{ hideCover?: boolean }>`
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: ${props => (props.hideCover ? "0" : "1")};
  visibility: ${props => (props.hideCover ? "hidden" : "visible")};
  transition: opacity 0.25s ease, visibility 0.25s ease;
`
