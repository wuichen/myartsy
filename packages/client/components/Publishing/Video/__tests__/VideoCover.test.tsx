import { VideoArticle } from "Components/Publishing/Fixtures/Articles"
import { Media } from "Components/Publishing/Fixtures/Components"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { VideoCover, VideoCoverAsset } from "../VideoCover"

describe("Video Cover", () => {
  it("matches the snapshot", () => {
    const videoCover = renderer
      .create(
        <VideoCover
          article={VideoArticle}
          media={Media[0]}
          seriesTitle="Future of Art"
        />
      )
      .toJSON()
    expect(videoCover).toMatchSnapshot()
  })

  it("matches the snapshot with edit props", () => {
    const videoInfo = renderer
      .create(
        <VideoCover
          article={VideoArticle}
          media={Media[0]}
          seriesTitle="Future of Art"
          editDescription={EditableChild("description")}
          editTitle={EditableChild("title")}
        />
      )
      .toJSON()
    expect(videoInfo).toMatchSnapshot()
  })

  it("renders video asset image", () => {
    const component = mount(
      <VideoCover
        article={VideoArticle}
        media={Media[0]}
        seriesTitle="Future of Art"
      />
    )
    expect(component.find(VideoCoverAsset).props().src).toEqual(
      "https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg"
    )
  })

  it("renders video info", () => {
    const component = mount(
      <VideoCover
        article={VideoArticle}
        media={Media[0]}
        seriesTitle="Future of Art"
      />
    )
    expect(component.text()).toMatch("The elegant spiral of the Nautilus shell")
    expect(component.text()).toMatch("Trevor Paglan")
  })

  it("renders editable fields", () => {
    const component = mount(
      <VideoCover
        article={VideoArticle}
        media={Media[0]}
        seriesTitle="Future of Art"
        editDescription={EditableChild("description")}
        editTitle={EditableChild("title")}
      />
    )
    expect(component.text()).toMatch("Child description")
    expect(component.text()).toMatch("Child title")
  })
})
