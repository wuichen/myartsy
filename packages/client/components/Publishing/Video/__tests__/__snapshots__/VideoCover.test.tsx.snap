// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Cover matches the snapshot 1`] = `
.c4 {
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 12px;
}

.c6 {
  padding-right: 20px;
  width: 60px;
}

.c13 {
  max-width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  margin-right: 20px;
}

.c11 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
}

.c12 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 42px;
  line-height: 50px;
}

.c14 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  padding-top: 30px;
}

.c8 {
  width: 32px;
  height: 32px;
}

.c9 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  background: url(https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg) no-repeat center center;
  background-size: cover;
  background-color: black;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background: linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.6));
  position: absolute;
  width: 100%;
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding-bottom: 40px;
  padding-left: 20px;
  padding-right: 20px;
  width: 100%;
  box-sizing: border-box;
  z-index: 1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c3 .c7 {
  height: 60px;
  width: 44px;
  cursor: pointer;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: 1;
  visibility: visible;
  -webkit-transition: opacity 0.25s ease,visibility 0.25s ease;
  transition: opacity 0.25s ease,visibility 0.25s ease;
}

@media screen and (min-width:40em) {
  .c13 {
    max-width: 60%;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:52em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:52em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c3 {
    padding-bottom: 60px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    src="https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
    width="100%"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="c6"
          onClick={[Function]}
          width="60px"
        >
          <svg
            className="c7 c8"
            version="1.1"
            viewBox="0 0 40 56"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <polygon
                fill="white"
                points="0 0 0 56 39.3600006 28"
              />
            </g>
          </svg>
        </div>
        <div
          className=""
        >
          <div>
            <div
              className="c5"
            >
              <div
                className="c9 c10"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                Future of Art
              </div>
              <div
                className="c11"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                02:28
              </div>
            </div>
            <h1
              className="c12"
              fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
              fontSize="42px"
            >
              Trevor Paglan
            </h1>
          </div>
        </div>
      </div>
      <div
        className="c13"
      >
        <div
          className="c14"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "22px",
              "22px",
              "22px",
            ]
          }
        >
          The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Cover matches the snapshot with edit props 1`] = `
.c4 {
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 12px;
}

.c6 {
  padding-right: 20px;
  width: 60px;
}

.c13 {
  max-width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  margin-right: 20px;
}

.c11 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
}

.c12 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 42px;
  line-height: 50px;
}

.c14 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  padding-top: 30px;
}

.c8 {
  width: 32px;
  height: 32px;
}

.c9 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  background: url(https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg) no-repeat center center;
  background-size: cover;
  background-color: black;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background: linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.6));
  position: absolute;
  width: 100%;
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding-bottom: 40px;
  padding-left: 20px;
  padding-right: 20px;
  width: 100%;
  box-sizing: border-box;
  z-index: 1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c3 .c7 {
  height: 60px;
  width: 44px;
  cursor: pointer;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: 1;
  visibility: visible;
  -webkit-transition: opacity 0.25s ease,visibility 0.25s ease;
  transition: opacity 0.25s ease,visibility 0.25s ease;
}

@media screen and (min-width:40em) {
  .c13 {
    max-width: 60%;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:52em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c14 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:52em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c14 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c3 {
    padding-bottom: 60px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    src="https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
    width="100%"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="c6"
          onClick={[Function]}
          width="60px"
        >
          <svg
            className="c7 c8"
            version="1.1"
            viewBox="0 0 40 56"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <polygon
                fill="white"
                points="0 0 0 56 39.3600006 28"
              />
            </g>
          </svg>
        </div>
        <div
          className=""
        >
          <div>
            <div
              className="c5"
            >
              <div
                className="c9 c10"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                Future of Art
              </div>
              <div
                className="c11"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                02:28
              </div>
            </div>
            <h1
              className="c12"
              fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
              fontSize="42px"
            >
              <div>
                Child 
                title
              </div>
            </h1>
          </div>
        </div>
      </div>
      <div
        className="c13"
      >
        <div
          className="c14"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "22px",
              "22px",
              "22px",
            ]
          }
        >
          <div>
            Child 
            description
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
